scripts/lock-patcher.mjs

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+/*
+  This node.js script is required in order to patch package-lock.json with new sha512 hashes.
+
+  It loads all `package-lock.json` files that are in root of the project as well as these
+  nested inside folders and patches their integrity based on packages snapshot created by Nix.
+
+  Sadly this can't be done with Nix due to restricted evaluation mode.
+  This script should not have any external npm dependencies
+*/
+
+import fsPromises from "fs/promises"
```

Comment:
i'd recommend naming it just `fs` but it does not really matter

Reply:
I would rather stick to `fsPromises` to make it more clear that it is not classic `fs` module.